snapcraft/internal/errors.py

```diff
@@ -39,8 +39,9 @@ def get_exit_code(self):
 
 class MissingStateCleanError(SnapcraftError):
     fmt = (
-        "Failed to clean step {step!r}: Missing necessary state. This won't "
-        "work until a complete clean has occurred."
+        "Failed to clean step {step!r}: "
+        "Missing necessary state. "
+        "This won't work until a complete clean has occurred."
```

Comment:
The sentence sounds like a factual statement to me whereas it should be advice on how to proceed. How about  "Run 'snapcraft clean' to start off fresh.".

Reply:
That's for the next PR, where I will change the "how to fix" part for snapcraft commands.

Reply:
More detail is better, but “Missing state” might be better than “State missing” because it avoids being misinterpreted as a command to the user.